### ISSUE TITLE
Problem with assigning certain weather files into models

### ISSUE THREAD

**Issue by ranchhand6** [NREL/OpenStudio#1748]
We've discovered that with about 16% of the weather models in a four state area (OR, WA, ID, MT), we get the following error when we try to use the change_building_location measure:

/private/var/folders/vs/c6y5rcmr8xj0yy0059b64z5h0000gr/T/OpenStudio.A86965/resources/run/1-UserScript-0/user_script.rb:108:in `setAnnualAverageOutdoorAirTemperature': Wrong arguments for overloaded method 'SiteWaterMainsTemperature.setAnnualAverageOutdoorAirTemperature'. (ArgumentError)
Possible C/C++ prototypes are:
    bool SiteWaterMainsTemperature.setAnnualAverageOutdoorAirTemperature(double annualAverageOutdoorAirTemperature)
    bool SiteWaterMainsTemperature.setAnnualAverageOutdoorAirTemperature(openstudio::Quantity const &annualAverageOutdoorAirTemperature)
    from /private/var/folders/vs/c6y5rcmr8xj0yy0059b64z5h0000gr/T/OpenStudio.A86965/resources/run/1-UserScript-0/user_script.rb:108:in`run'
    from in.rb:325:in `<main>'

The weather file that caused this error is:  USA_WA_Renton.Muni.AP.727934_TMY3.epw
A weather file that works is:  USA_WA_Hanford.727840_TMY3.epw

I can provide a full list of all the weather files we tested and their respective results if that helps.

This happens in both the OpenStudio desktop software (V1.8.0, build  97b69c8a8d) and in the OpenStudio-server software (1.11.0-rc3).

I suspect that this an issue unrelated to the model or the measure because they do work in most cases, but if there's a switch we've missed, I'd be more than happy to know about it.

As a temporary measure we've removed the offending weather files from our project, but we don't consider this a long-term solution.

Thanks for all the hard work!

**Comment by DavidGoldwasser** [comment]
@macumber has issues with bad weather files, and may know what the problem is.

**Comment by macumber** [comment]
@DavidGoldwasser that looks like a problem with that measure and not the general EpwFile parsing

@jasondegraw FYI

**Comment by macumber** [comment]
Assigning back to @DavidGoldwasser since this is measure specific

**Comment by ranchhand6** [comment]
If this is in the measure, could someone give me a clue as to how to fix the measure so that it works with the OpenStudio software?  Seems to me like the error happens during the assignment of the weather file to the simulation, which is below the level that the measure can check.  Am I just not seeing something?

**Comment by macumber** [comment]
I am not familiar with this measure (@DavidGoldwasser wrote it) but it looks like it is trying to parse a stat file to get the mean dry bulb temperature:

https://github.com/NREL/OpenStudio-measures/blob/develop/NREL%20working%20measures/ChangeBuildingLocation/measure.rb#L132

Either that weather file does not have a stat file, the stat file is incorrect, or the stat file parser in that measure is incorrect:

https://github.com/NREL/OpenStudio-measures/blob/develop/NREL%20working%20measures/ChangeBuildingLocation/resources/stat_file.rb

**Comment by DavidGoldwasser** [comment]
@ranchhand6 are you downloading the weather data from here
http://apps1.eere.energy.gov/buildings/energyplus/cfm/weather_data3.cfm/region=4_north_and_central_america_wmo_region_4/country=1_usa/cname=USA

If so I can take the two files you mention in original question (one that works and one that doesn't) and I can try and work it out. I susspect @macumber is right about stat file. If the issue is invalid or missing stat file I should issue a warning about not being able to calculate water temperatures but still set the weather file.

**Comment by ranchhand6** [comment]
Yes, I believe so.  We downloaded the files about a year ago and checked them into a deployment repository.  We haven't updated them recently, I can try that if you think it might be an issue.

**Comment by DavidGoldwasser** [comment]
I'll let you know if I can't reproduce the error with what I pull off the site.

**Comment by ranchhand6** [comment]
Any updates on this one?

**Comment by DavidGoldwasser** [comment]
Sorry, I haven't resolved this yet, although I should point out that measures that change weather files won't work properly in the OS app or PAT when done at RunTime. You can run them in Apply measures now and on OpenStudio server.

**Comment by ranchhand6** [comment]
Hmm, Ok, we didn't know that.  The OpenStudio server case is the one we're interested in, in any case.

**Comment by macumber** [comment]
@nllong @henryHorsey @brianlball FYI

**Comment by brianlball** [comment]
@ranchhand6, can you try running this again with server version 1.12.5 or higher?  This looks similiar to the regex issue we found in E+

**Comment by DavidGoldwasser** [comment]
@ranchhand6 I updated this measure to add a lot of runner logs. I it didn't have initial and finial condition, and it didn't have "return false" after errors. Here is the updated measure.
https://bcl.nrel.gov/node/82184

I then added a test using the USA_WA_Renton.Muni.AP.727934_TMY3 that runs fine. I did however notice that the .ddy file for this location doesn't have any design days, just looks like this.

```
 Site:Location,
  Renton Muni_WA_USA Design_Conditions,     !- Location Name
      47.48,     !- Latitude {N+ S-}
    -122.22,     !- Longitude {W- E+}
      -8.00,     !- Time Zone Relative to GMT {GMT+/-}
      10.00;     !- Elevation {m}


 ! No Design Conditions found for this Location
```

**Comment by ranchhand6** [comment]
Excellent, thanks, we'll give it a go!

**Comment by ranchhand6** [comment]
So, this is not working for some reason.  I'm not sure that I've got the right version.  When I click on the node in the link, I get a slightly changed measure.rb:

diff measures/ChangeBuildingLocation/measure.rb measures/ChangeBuildingLocation_old/measure.rb 
127c127
<         ddy_list = /(Htg 99.6. Condns DB)|(Clg .4. Condns WB=>MDB)|(Clg .4% Condns DB=>MWB)/
---
>         ddy_list = /(Htg 99.6. Condns DB)|(Clg .4. Condns WB=>MDB)|(Clg .4% Condns DB=>MWB)|(Hum_n 99.6% Condns DP=>MCDB)|(Htg Wind 99.6% Condns WS=>MCDB)|(Clg .4% Condns DP=>MDB)|(Clg .4% Condns Enth=>MDB)/

This seems to be fewer changes than you described in your text.  Also, I don't see the additional weather files in the tests folder, which I thought might be there.

Does the measure website have a batch update cycle that I need to wait for?

**Comment by DavidGoldwasser** [comment]
There could be some delay. You will have to go and update the measure manually in the app. It should see the update and offer to let you download an updated copy.
https://unmethours.com/question/4762/how-do-i-know-if-any-of-my-bcl-measures-or-components-have-updates/

You should see a new argument for climate zone in the arguments section and code that sets the climate zone in the measure.rb file

```ruby     # Set climate zone
    climateZones = model.getClimateZones
    climateZones.setClimateZone("ASHRAE",climate_zone)
```
You should also see a new test ' def test_weather_file_WA_Renton' in the change_building_location_test.rb file. I do see that the new weather files were stripped out on copy uploaded to BCL. For some reason measure.xml didn't update and identify the new test files. This shouldn't affect use of the measure, but just the ability to run the test.

**Comment by ranchhand6** [comment]
OpenStudio doesn't seem to recognize an updated measure in the BCL.  I'll give it 24 hours and try again.

**Comment by ranchhand6** [comment]
Could you please double-check that the right version of the software is available on the BCL website?  There are no differences between the version downloaded today versus yesterday and neither of them have the code snippet from your previous comment.

**Comment by ranchhand6** [comment]
Perhaps you could tar up a version of this measure and send it to me in email?  Still no change on the website.  Thanks!

**Comment by DavidGoldwasser** [comment]
Sure, send me your email to OpenStudio@NREL.gov and I'll send it to you, but just for reference, what happens if you try to download it from here directly from the website vs. downloading it from within the Application. Odd that you are not seeing the update.

**Comment by ranchhand6** [comment]
So, we're trying to run the measure against our OpenStudio server and are getting the following error:

[23:13:44.865670 ERROR] Found error in state 'energyplus' with message ["/opt/rbenv/versions/2.0.0-p481/lib/ruby/gems/2.0.0/bundler/gems/OpenStudio-workflow-gem-f478f4e96700/lib/openstudio/workflow/jobs/run_energyplus/run_energyp
lus.rb failed with EnergyPlus Terminated with a Fatal Error. Check eplusout.err log., 

Checking eplusout.err:

$ more eplusout.err
Program Version,EnergyPlus, Version 8.2.0-8397c2e30b, YMD=2015.10.08 23:13,IDD_Version 8.2.0
   ** Severe  ** IP: IDF line~29304 Object=CONTROLLER:MECHANICALVENTILATION, name=CONTROLLER MECHANICAL VENTILATION 1, Required Field=[Zone 1 Name] was blank.
   ** Warning ** IP: Note -- Some missing fields have been filled with defaults. See the audit output file for details.
   ** Severe  ** IP: Possible incorrect IDD File
   **   ~~~   ** IDD Version:"IDD_Version 8.2.0"
   **   ~~~   ** Version in IDF="8.2" not the same as expected="8.2"
   **   ~~~   ** Possible Invalid Numerics or other problems
   **  Fatal  ** IP: Errors occurred on processing IDF file. Preceding condition(s) cause termination.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=2
   ..... Last severe error=IP: Possible incorrect IDD File
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 1 Warning; 2 Severe Errors; Elapsed Time=00hr 00min  1.00sec

It does look like the measure applied correctly, there are no errors in the log at that point.

As a guess, perhaps we need to update our server?  Does that look plausible?

I can send you the error logs if that would be helpful.

**Comment by brianlball** [comment]
what version of the os-server are you using?
the latest is 1.12.13 which has OS 1.9.1.

**Comment by ranchhand6** [comment]
Looks like it's a few back:

Version of OpenStudio Server: 1.11.0-rc3

It think it's OS 1.8.0 or so.

**Comment by brianlball** [comment]
can you try with 1.12.13?

**Comment by ranchhand6** [comment]
I'll give it a go tomorrow.

**Comment by ranchhand6** [comment]
Doesn't look like it worked.  Here's the error:

ubuntu@ip-10-170-154-160:/mnt/openstudio/analysis_4bc2ae74-0989-4721-ad62-8902b6332cb1/data_point_704c705c-e635-4b6a-9457-7ab8c6e34a41/run$ more eplusout.err 
Program Version,EnergyPlus, Version 8.3.0-6d97d074ea, YMD=2015.10.12 16:46,IDD_Version 8.3.0
   ** Severe  ** IP: IDF line~29315 Object=CONTROLLER:MECHANICALVENTILATION, name=CONTROLLER MECHANICAL VENTILATION 1, Required Field=[Zone 1 Name] was blank.
   ** Warning ** IP: Note -- Some missing fields have been filled with defaults. See the audit output file for details.
   ** Severe  ** IP: Possible incorrect IDD File
   **   ~~~   ** IDD Version:"IDD_Version 8.3.0"
   **   ~~~   ** Version in IDF="8.3" not the same as expected="8.3"
   **   ~~~   ** Possible Invalid Numerics or other problems
   **  Fatal  ** IP: Errors occurred on processing IDF file. Preceding condition(s) cause termination.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=2
   ..... Last severe error=IP: Possible incorrect IDD File
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 1 Warning; 2 Severe Errors; Elapsed Time=00hr 00min  0.56sec

Here's the version:  Version of OpenStudio Server: 1.12.4

I can send you the whole datapoint if that helps.

**Comment by ranchhand6** [comment]
Hmm.  Just noticed that the server version wasn't 1.12.13.  I'm using OpenStudio-spreadsheet to generate the servers.  Let me check my other methods to see if I have access to 1.12.13.

**Comment by ranchhand6** [comment]
I am unable to test with 1.12.13.  I tried doing a 'git pull' on OpenStudio-server and going a 'vagrant up' on the server subdirectory, but the chef recipe didn't complete correctly.  I spent a little time trying to troubleshoot it but came up empty.

So, I downloaded the latest version of PAT and used it to throw a set of servers up into AWS.  This registered as 'Version of OpenStudio Server: 1.12.12', which I thought was close enough to test.  Unfortunately, neither PAT nor my test scripts could get a simulation to start on this infrastructure.  The overall job is marked "started" but the individual job is marked "queued" and never progresses.  I've seen this before in servers and workers and it normally is a symptom of the security group getting set too tightly, but I didn't go down that path.  Instead, I tried to get PAT to set up a cloud group with 0 workers, reasoning that this would solve any network issues between the workers and the server.  Unfortunately, unlike OpenStudio-server-worksheet, PAT does not seem to be able to save 0 as a worker configuration choice.

So, this does not seem to work.  If I have some time in the evening, I may try to delete the vagrant instance and recreate it.

Any suggestions welcome.
